### ISSUE TITLE
"" is not treated as a const when passed to natives.

### ISSUE THREAD

**Issue by KyleSanderson** [alliedmodders/sourcepawn#143]
#pragma semicolon 1
#include <sourcemod>

/* native bool:GetTrieString(Handle:map, const String:key[], String:value[], max_size, &size=0) */

public OnPluginStart()
{
	PrintToServer("%b", GetTrieString(INVALID_HANDLE, "pickles", "", 0));
}

/*
Your plugin successfully compiled!
Use the link below to download your plugin. It will be valid for ten minutes.

http://www.sourcemod.net/compiler.php?go=dl&id=685667
SourcePawn Compiler 1.7.1
Copyright (c) 1997-2006 ITB CompuPhase
Copyright (c) 2004-2014 AlliedModders LLC

Code size:             3044 bytes
Data size:             2240 bytes
Stack/heap size:      16384 bytes
Total requirements:   21668 bytes
*/

**Comment by dvander** [comment]
Indeed, a little known fact: string literals are not constant. Probably by mistake more than design, but one case where it matters is default arguments. The compiler does not coalesce redundant strings, meaning:

```
GetTrieString(INVALID_HANDLE, "pickles", "", 0)
GetTrieString(INVALID_HANDLE, "pickles", "", 0)
```

Will generate two separate memory locations each containing "", and if one gets overwritten due to malformed code, the other won't be affected.
